grim.go

```diff
@@ -248,3 +250,27 @@ Or to use the server defaults, remove the entry %q`
 func buildTruncatedMessage(truncateID string) string {
 	return fmt.Sprintf(truncatedMessage, truncateID, truncateID, truncateID)
 }
+
+func shouldSkip(hook *hookEvent) *string {
+	var message *string
+
+	switch {
+	case hook.Deleted:
+		message = getStringPtr("because it was on a deleted branch")
+	case hook.EventName == "push":
+	case hook.EventName == "pull_request":
```

Comment:
exceeding nitpick (dont feel like you have to do this), but once you go down into a nested switch statement, it probably make sense to have another function "shouldSkipPullRequest" that does the inner branch.